lib/nats.js

```diff
@@ -401,7 +444,7 @@ Client.prototype.setupHandlers = function() {
     // Only bubble up error if we never had connected
     // to the server and we only have one.
     if (client.wasConnected === false && client.servers.length === 0) {
-      client.emit('error', 'Could not connect to server: ' + exception);
+      client.emit('error', new NatsError(CONN_ERR_MSG_PREFIX + exception, CONN_ERR, exception));
```

Comment:
Seems that the `chainedError` is only used for this case, maybe an alternative could be to just thrown the error opaquely instead of wrapping with new NatsError type? Think that way caller could catch the original type and nats errors without having to check further state on the nats error which was thrown.